src/Synth/ADnote.cpp

```diff
@@ -420,22 +467,112 @@ void ADnote::setupVoiceDetune(int nvoice)
                 pars.VoicePar[nvoice].PFMDetune);
 }
 
+/*
+    There are multiple places where the modulation mode ("FM mode") is being stored:
+    - ADnoteVoiceParam::PFMenabled (var. "param")
+    - ADnoteVoiceParam::table::mode()
+    - NoteVoicePar::FMEnabled (var. "voice")
+
+    Assume they are all in sync. Here is how they change:
+
+    User sends OSC message ".../PFMenabled" -> sets PFMenabled
+       ...
+    Wavetable is regenerated
+       ...
+    New wavetable is ready : WaveTable::mode is changed
+       ...
+    ADnote takes FMEnabled to match WaveTable::mode (this function)
```

Comment:
This sequencing is written out like a perfect integration test case :)

Reply:
Reading the below code it sounds like a fair bit of testing has been done on the scenario:

- Start note
- change modulation
- let wavetables regen
- change modulation
- let wavetables regen
- stop note

It's less clear if there's been tests around:

- change modulation
- start note
- let wavetables regen
- stop note

Did you target the later scenario? i.e. first_run=true, wtModHasChanged=true